microscope/monitor/parser.py

```diff
@@ -81,68 +81,211 @@ def pop_current(self, init: str="") -> str:
+            return self.parse_debug(event)
+        if event["type"] == "capture":
+            return self.parse_capture(event)
+        if event["type"] == "agent":
+            return self.parse_agent(event)
+
+        return e
+
+    def parse_labels(self, labels: List[str]) -> str:
+        return ", ".join([l for l in labels
+                          if "k8s:io.kubernetes.pod.namespace=" not in l])
+
+    def parse_l7(self, event: Dict) -> str:
+        src_labels = self.parse_labels(event["srcEpLabels"])
+        dst_labels = self.parse_labels(event["dstEpLabels"])
+
+        action = ""
+        if "http" in event:
+            http = event['http']
+            action = f"{http['Method']} {http['URL']['Path']}"
+
+        if "kafka" in event:
+            kafka = event['kafka']
+            action = f"{kafka['APIKey']} {kafka['Topic']['Topic']}"
+
+        return (f"({src_labels}) => ({dst_labels}) {event['l7Proto']}"
+                f" {action} {event['verdict']}")
+
+    def parse_trace(self, event: Dict) -> str:
+        src_ep, dst_ep = self.get_eps_repr(event)
+
+        return (f"trace ({src_ep}) =>"
+                f" ({dst_ep})")
+
+    def parse_drop(self, event: Dict) -> str:
+        src_ep, dst_ep = self.get_eps_repr(event)
+
+        return (f"drop: {event['reason']} ({src_ep}) =>"
+                f" ({dst_ep})")
+
+    def parse_debug(self, event: Dict) -> str:
+        return f"debug: {event['message']} on {event['cpu']}"
 
-    def is_full(self, line: str) -> bool:
-        return ("=>" in line or "Listening for events" in line
-                or "Press Ctrl-C" in line)
+    def parse_capture(self, event: Dict) -> str:
+        return f"{event['prefix']}: {event['summary']}"
+
+    def parse_agent(self, event: Dict) -> str:
+        return f"{event['subtype']}: {event['message']}"
+
+    def get_eps_repr(self, event: Dict) -> Tuple[str, str]:
+        """
+        get_eps_repr returns tuple with source endpoint
+        and destination endpoint representation
+        """
+        src_repr = ""
+        dst_repr = ""
+        src_ip = ""
+        dst_ip = ""
+        src_port = ""
+        dst_port = ""
+        src_ip_l4 = ""
+        dst_ip_l4 = ""
+
+        try:
+            src_ip, dst_ip = self.get_ips(event)
+        except (KeyError, StopIteration):
+            pass
+        try:
+            src_ip, dst_ip = self.get_ips(event)
+        except KeyError:
+            pass
+
+        try:
+            src_port, dst_port = self.get_ports(event)
+        except (KeyError, StopIteration):
+            pass
+
+        if src_port and src_ip:
+            src_ip_l4 = src_ip + ":" + src_port
+
+        if dst_port and dst_ip:
+            dst_ip_l4 = dst_ip + ":" + dst_port
+
+        try:
+            if src_ip:
+                src_ep = self.resolver.resolve_ip(src_ip)
+                src_repr = (
+                    f"{src_ep}"
+                )
+        except (KeyError, StopIteration):
```

Comment:
Since we do these on each packet, multiple times, is the exception handling faster than a `if key in dict` style check?

Reply:
Exception handling is slightly faster if the key exists, but slower if the key is not there. I will change this to if checks.